src/main/java/frc/robot/util/Node.java

```diff
@@ -0,0 +1,67 @@
+package frc.robot.util;
+
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import frc.robot.RobotContainer;
+
+/**Class to store data about scoring locations*/
+public class Node {
+    // Possible node types
+    public enum Types {CONE, CUBE, HYBRID};
+
+    public final Alliance alliance;
+    public final int row;
+    public final int x;
+    public final Pose3d pose;
+    public final Pose2d scorePose;
+    public final Types type;
+
+    /**
+     * Creates a new Node object
+     * @param team
+     *  The node's color
+     * @param row
+     *  Which row it's in (1 = bottom, 2 = middle, 3 = top)
+     * @param x
+     *  Grid column from left to right (1-9)
+     */
+    public Node(Alliance alliance, int row, int x){
+        this.alliance=alliance;
+        this.row=row;
+        this.x=x;
+        type = row==1?Types.HYBRID:x%3==2?Types.CUBE:Types.CONE;
+
+        // Which April tag this node is closest to
+        Pose3d tag = RobotContainer.aprilTags[(x-1)/3+(alliance==Alliance.Red?1:6)];
```

Comment:
Why not just save the tag number? We do not need to store Pose3D values in many places.

Reply:
Again, it avoids any possible error if we ever want to do something like aprilTags[1]==Vision.getTagPose(1). If they're all from the array, they will always reference the same variable.